syntax = "proto3";
option objc_class_prefix = "DRS";
package huawei.proto;

service Replicator {
    rpc replicate(stream ReplicateRequest) returns(stream ReplicateResponse) {}
    rpc sync_marker(ReplicateRequest) returns(ReplicateResponse){}
}
message ReplicateRequest {
    int64 id = 1;
    CMD_TYPE cmd = 2;
    ENCODE_TYPE encode_type = 3;
    string vol_id = 4;
    int32 state = 5; //state of replicating journal:opened/sealed?
    int64 current_counter = 6; //the replicating journal counter
    int32 offset = 7;
    bytes data = 8;
}
message ReplicateResponse {
    int64 id = 1;
    int32 res = 2;
}
enum CMD_TYPE {
	UNKNOWN_CMD = 0;
	START_CMD = 1;
	DATA_CMD = 2;
	FINISH_CMD = 3;
}
enum ENCODE_TYPE {
    UNKNOWN_EN = 0;
    NONE_EN = 1;
}